src/org/traccar/model/ExtendedModel.java

```diff
@@ -108,14 +108,18 @@ public boolean getBoolean(String key) {
         }
     }
 
-    public int getInteger(String key) {
+    public Integer getInteger(String key, Integer defaultValue) {
```

Comment:
What is this method for?

Reply:
how do you detect that an integer attribute is not present? See in MainEventHandler for example.

Reply:
Usually we check `containsKey`.

Reply:
yeah, but here you have let's say a Device d, then you would need: d.getAttributes().containsKey('foo'), if it was d.containsKey('foo') then I would maybe agree.

Reply:
If you want to change API here, it should probably be the other way around. Single argument can return null, but one with default value shouldn't. And we should use it across the app. I don't want to implement a new API just for logging.

Reply:
or another way we can do: rename (refactor) the current function to: int getInt(String key), and add a new function: Integer getInteger(String key) that will return null if it's not contained.

Reply:
I don't see the point of creating a function just to use it in a single place. If you want to change everything to make it consistent, you are welcome, but probably in different PR. For now I would recommend to stick to existing practice.

Reply:
ok